### PR TITLE
gf-complete: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

**Comment by valgur** [comment]
@uilianries @RubenRBS `ERROR: Missing prebuilt package for 'm4/1.4.19'`

File: recipes/gf-complete/all/conanfile.py [hunk 1]
```diff
@@ -1,162 +1,193 @@
...
+            # Custom AutotoolsDeps for cl like compilers
+            # workaround for https://github.com/conan-io/conan/issues/12784
+            includedirs = []
+            defines = []
+            libs = []
+            libdirs = []
+            linkflags = []
+            cxxflags = []
+            cflags = []
+            for dependency in self.dependencies.values():
```
Comment (RubenRBS) [inline recipes/gf-complete/all/conanfile.py:152]: Does this maybe need to be in topological order, like @SpaceIm has been doing in a few similar cases?

**Review by RubenRBS — COMMENTED** [review]
Thanks! One question about the autotoolsdeps workaround, else looks good :)

File: recipes/gf-complete/all/conanfile.py [hunk 2]
```diff
@@ -1,162 +1,198 @@
...
             if self.options.avx != "auto":
-                conf_args.append("--enable-avx={}".format(yes_no(self.options.avx)))
-
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-
-        return self._autotools
+                tc.configure_args.append("--enable-avx={}".format(yes_no(self.options.avx)))
+        if is_msvc(self):
+            if self.options.shared:
+                tc.extra_ldflags.append("-no-undefined")
```
Comment (RubenRBS) [inline recipes/gf-complete/all/conanfile.py:125]: Any insight into why this one is needed?

**Review by RubenRBS — COMMENTED** [review]
Question about the new msvc shared build code, otherwise looks great, thanks a lot for your patience @valgur

File: recipes/gf-complete/all/conanfile.py (hunk 2 above)
Comment (valgur) [reply recipes/gf-complete/all/conanfile.py:125]: It's been a while, but I think I added it as an Autotools substitute for `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS`.
The `-no-undefined` was probably just to be safe.

**Review by RubenRBS — APPROVED** [review]

File: recipes/gf-complete/all/conanfile.py [hunk 3]
```diff
@@ -1,162 +1,198 @@
...
+        with chdir(self, self.source_folder):
+            autotools = Autotools(self)
             autotools.install()
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+        rm(self, "*.la", self.package_folder, recursive=True)
+        fix_apple_shared_install_name(self)
 
     def package_info(self):
-        self.cpp_info.libs = ["gf_complete"]
+        self.cpp_info.libs = collect_libs(self)
```
Comment (uilianries) [inline recipes/gf-complete/all/conanfile.py:193]: Please, avoid collect_libs. It's hard to predict what's expected, plus, it does not respect link order.

**Review by jwillikers — APPROVED** [review]

**Pull request merged**